sdk/go/kubernetes/yaml/yaml.go

```diff
@@ -0,0 +1,1800 @@
+		"extensions/v1beta1/DaemonSetList",
+		"extensions/v1beta1/DeploymentList",
+		"extensions/v1beta1/IngressList",
+		"extensions/v1beta1/NetworkPolicyList",
+		"extensions/v1beta1/PodSecurityPolicyList",
+		"extensions/v1beta1/ReplicaSetList",
+		"flowcontrol.apiserver.k8s.io/v1alpha1/FlowSchemaList",
+		"flowcontrol.apiserver.k8s.io/v1alpha1/PriorityLevelConfigurationList",
+		"networking.k8s.io/v1/NetworkPolicyList",
+		"networking.k8s.io/v1beta1/IngressClassList",
+		"networking.k8s.io/v1beta1/IngressList",
+		"node.k8s.io/v1alpha1/RuntimeClassList",
+		"node.k8s.io/v1beta1/RuntimeClassList",
+		"policy/v1beta1/PodDisruptionBudgetList",
+		"policy/v1beta1/PodSecurityPolicyList",
+		"rbac.authorization.k8s.io/v1/ClusterRoleBindingList",
+		"rbac.authorization.k8s.io/v1/ClusterRoleList",
+		"rbac.authorization.k8s.io/v1/RoleBindingList",
+		"rbac.authorization.k8s.io/v1/RoleList",
+		"rbac.authorization.k8s.io/v1alpha1/ClusterRoleBindingList",
+		"rbac.authorization.k8s.io/v1alpha1/ClusterRoleList",
+		"rbac.authorization.k8s.io/v1alpha1/RoleBindingList",
+		"rbac.authorization.k8s.io/v1alpha1/RoleList",
+		"rbac.authorization.k8s.io/v1beta1/ClusterRoleBindingList",
+		"rbac.authorization.k8s.io/v1beta1/ClusterRoleList",
+		"rbac.authorization.k8s.io/v1beta1/RoleBindingList",
+		"rbac.authorization.k8s.io/v1beta1/RoleList",
+		"scheduling.k8s.io/v1/PriorityClassList",
+		"scheduling.k8s.io/v1alpha1/PriorityClassList",
+		"scheduling.k8s.io/v1beta1/PriorityClassList",
+		"settings.k8s.io/v1alpha1/PodPresetList",
+		"storage.k8s.io/v1/CSIDriverList",
+		"storage.k8s.io/v1/CSINodeList",
+		"storage.k8s.io/v1/StorageClassList",
+		"storage.k8s.io/v1/VolumeAttachmentList",
+		"storage.k8s.io/v1alpha1/VolumeAttachmentList",
+		"storage.k8s.io/v1beta1/CSIDriverList",
+		"storage.k8s.io/v1beta1/CSINodeList",
+		"storage.k8s.io/v1beta1/StorageClassList",
+		"storage.k8s.io/v1beta1/VolumeAttachmentList":
+		var resources []resourceTupleOutput
+		if rawItems, hasItems := obj["items"]; hasItems {
+			if items, ok := rawItems.([]interface{}); ok {
+				for _, item := range items {
+					if obj, ok := item.(map[string]interface{}); ok {
+						rs, err := parseYamlObject(ctx, obj, transformations, resourcePrefix, opts...)
+						if err != nil {
+							return nil, err
+						}
+						for _, r := range rs {
+							resources = append(resources, r)
+						}
+					}
+				}
+			}
+		}
+		return resources, nil
+	}
+
+	// If we got here, it's not a recursively traversed type, so process it directly.
+	// First, validate that it has the requisite metadata and name properties.
+	meta, hasMeta := obj["metadata"]
+	if !hasMeta {
+		return nil, errors.Errorf("YAML object does not have a .metadata field: %s %+v", fullKind, obj)
+	}
+	metaDict, hasMetaDict := meta.(map[string]interface{})
+	if !hasMetaDict {
+		return nil, errors.Errorf("YAML object does not have a .metadata dictionary: %s %+v", fullKind, obj)
+	}
+	metaName, hasMetaName := metaDict["name"].(string)
```

Comment:
This type test can fail if the name is an `Output`.  If we want to match the TypeScript behaviour we should support this being an output and then do the `RegisterResource` below in an `apply` off this output, like in the Node.js code.

If we decide not to support that for now (and error on it like we do here), then we could likely simplify this code substantially (there is no need for any outputs).  However, note that it might be a breaking change to add support for this later if we don't include an `Output` in the public API now - so we may still want to inject an `Output` wrapper in the public API where these things get exposed so that we have the option to support this in the future as in TypeScript.

Reply:
After further investigation, we decided not to support `name` or `namespace` being Outputs in the Go SDK. The args only accept `strings`, so the only way this can happen is through a transformation. This is most commonly seen if a transformation is used to set resource namespaces (e.g., https://github.com/pulumi/pulumi-kubernetes/issues/478).

I opened https://github.com/pulumi/pulumi-kubernetes/issues/1101 to track a better way of handling the namespace case.